packages/ai/core/generate-text/stream-text.ts

```diff
@@ -723,6 +760,44 @@ class DefaultStreamTextResult<TOOLS extends Record<string, CoreTool>>
                   ? JSON.stringify(stepToolCalls)
                   : undefined;
 
+              // determine the next step type
+              let nextStepType: 'done' | 'continue' | 'tool-result' = 'done';
+              if (currentStep + 1 < maxSteps) {
+                if (
+                  continueSteps &&
+                  stepFinishReason === 'length' &&
+                  // only use continue when there are no tool calls:
+                  stepToolCalls.length === 0
+                ) {
+                  nextStepType = 'continue';
+                } else if (
+                  // there are tool calls:
+                  stepToolCalls.length > 0 &&
+                  // all current tool calls have results:
+                  stepToolResults.length === stepToolCalls.length
+                ) {
+                  nextStepType = 'tool-result';
+                }
+              }
+
+              // when using continuation, publish buffer on final step or if there
+              // was no whitespace in the step:
+              if (
+                continueSteps &&
+                chunkBuffer.length > 0 &&
+                ((stepType === 'continue' && nextStepType === 'done') ||
+                  (stepType === 'continue' && !chunkTextPublished))
```

Comment:
bug: what if continueSteps is true, there is remaining buffer, but not continue step (just initial)